### PR TITLE
Feature/catalogue features

### PR DESCRIPTION
**Pull request by AltysS**

### CODE REVIEW

**Review by AlinaPalyvoda — APPROVED** [review]
I have no comments. I agree with everything. Great job!

File: client/src/store/index.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { configureStore } from '@reduxjs/toolkit';
 import userReducer from './slices/userSlice';
+import catalogueReducer from './slices/catalogueSlice';
 
 const store = configureStore({
-  reducer: { userReducer },
+  reducer: { userReducer, catalogueReducer },
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/index.js]:
```suggestion
  reducer: { userReducer, 
  catalogue: catalogueReducer },
```

File: client/src/api/getProducts.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import axiosConfig from '../axiosConfig';
+
+const getProducts = async () => {
+  try {
+    const { data } = await axiosConfig.get('/products');
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/api/getProducts.js]:
```suggestion
    const { data } = await axios('/products');
```

simpler

File: client/src/pages/CataloguePage/CataloguePage.js [hunk 3]
```diff
@@ -16,29 +19,62 @@ const FilterContainer = styled((props) => <Grid item xs={12} {...props} />)(({ t
...
-            <CatalogTourCard />
-            <CatalogTourCard />
-            <CatalogTourCard />
-            <CatalogTourCard />
-          </Stack>
-        </Grid>
+const CataloguePage = () => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(null);
+  const products = useSelector((state) => state.catalogueReducer.products, shallowEqual);
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/pages/CataloguePage/CataloguePage.js]:
```suggestion
  const products = useSelector((store) => store.catalogue.products, shallowEqual);
```

File: client/src/pages/CataloguePage/CataloguePage.js [hunk 4]
```diff
@@ -16,29 +19,62 @@ const FilterContainer = styled((props) => <Grid item xs={12} {...props} />)(({ t
...
+      const data = await getProducts();
+      dispatch(setProducts(data));
+      setLoading(false);
+    };
+    if (products.length > 0) {
+      setLoading(false);
+    } else {
+      getData();
+    }
+  }, []);
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/pages/CataloguePage/CataloguePage.js]: this is not correct,   we are using Redux toolkit.  the async request has to be encapuslated in the Slice file.  and just called in here.    you can check my tourSlice as an example. you will find a THUNK function called fetchTour

isLoading logic also has to be encapsulated into the Redux thunk function.

**Review by Ahmed-Fahmy-Dev — CHANGES_REQUESTED** [review]
as a final comment, the file structure in the store is wrong.  i guess you took Roman's code as an example.   you have to create a folder for each slice, where the slice itself will be located plus the test file.  

so the structure is as follows:       store/catalogueSlice/catalogueSlice.js

File: client/src/store/index.js (hunk 1 above)
Comment (AltysS) [reply client/src/store/index.js]: solved

Comment (AltysS) [reply client/src/store/index.js]: sovled

File: client/src/api/getProducts.js (hunk 2 above)
Comment (AltysS) [reply client/src/api/getProducts.js]: solved

File: client/src/pages/CataloguePage/CataloguePage.js (hunk 3 above)
Comment (AltysS) [reply client/src/pages/CataloguePage/CataloguePage.js]: solved

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]
approved for setIsLoading to be fixed by encapsulating itto getProducts function

**Pull request merged**